code/game/objects/items.dm

```diff
@@ -28,6 +28,8 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 	///Icon file for right inhand overlays
 	var/righthand_file = 'icons/mob/inhands/items_righthand.dmi'
 
+	///Override mob worn overlay with invisible overlay. Explicitly sets an item to be hidden when worn by displaying a pre-defined icon 'icons/misc/hidden.dmi' and icon_state "hidden" as defined in human_update_icons.dm.
+	var/worn_hidden = FALSE

```

Comment:
just make an icon called "hidden_icon" and set the worn_icon_state var to that. no need for a new var really.

Reply:
Yep, added the hidden icon and fixed neckerchief underlying layer with it. Var taken out and all other affected items as well.